---
cartesian_pose_interface:
    type: franka_interface_ros1/CartesianPoseInterface
    arm_id: panda

cartesian_vel_interface:
    type: franka_interface_ros1/CartesianVelInterface
    arm_id: panda
    alpha: 0.995

cartesian_state_interface:
    type: franka_interface_ros1/CartesianStateInterface
    arm_id: panda

joint_position_interface:
    type: franka_interface_ros1/JointPositionInterface
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

franka_state_controller:
  type: franka_control/FrankaStateController
  publish_rate: 100  # [Hz] default was 30
  joint_names:
    - panda_joint1
    - panda_joint2
    - panda_joint3
    - panda_joint4
    - panda_joint5
    - panda_joint6
    - panda_joint7
  arm_id: panda

franka_control:
  joint_names:
    - panda_joint1
    - panda_joint2
    - panda_joint3
    - panda_joint4
    - panda_joint5
    - panda_joint6
    - panda_joint7
  arm_id: panda
  # Configure the threshold angle for printing joint limit warnings.
  joint_limit_warning_threshold: 0.1 # [rad]
  # Activate rate limiter? [true|false]
  rate_limiting: true
  # Cutoff frequency of the low-pass filter. Set to >= 1000 to deactivate.
  cutoff_frequency: 100
  # Internal controller for motion generators [joint_impedance|cartesian_impedance]
  internal_controller: cartesian_impedance # joint_impedance
  # Used to decide whether to enforce realtime mode [enforce|ignore]
  realtime_config: enforce
  # Configure the initial defaults for the collision behavior reflexes. (default)
  # collision_config:
  #   lower_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   upper_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   lower_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   upper_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   lower_force_thresholds_acceleration: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
  #   upper_force_thresholds_acceleration: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
  #   lower_force_thresholds_nominal: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
  #   upper_force_thresholds_nominal: [20.0, 20.0, 20.0, 25.0, 25.0, 25.0]  # [N, N, N, Nm, Nm, Nm]
  # Configure the initial defaults for the collision behavior reflexes. (modified)
  collision_config:
      lower_torque_thresholds_acceleration: [20.0, 20.0, 20.0, 20.0, 20.0, 20.0, 20.0] # [Nm]
      upper_torque_thresholds_acceleration: [40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0] # [Nm]
      lower_torque_thresholds_nominal: [20.0, 20.0, 20.0, 20.0, 20.0, 20.0, 20.0] # [Nm]
      upper_torque_thresholds_nominal: [40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0] # [Nm]
      lower_force_thresholds_acceleration: [20.0, 20.0, 20.0, 20.0, 20.0, 20.0] # [N,N,N,Nm,Nm.Nm]
      upper_force_thresholds_acceleration: [40.0, 40.0, 40.0, 40.0, 40.0, 40.0]  # [N,N,N,Nm,Nm.Nm]
      lower_force_thresholds_nominal: [20.0, 20.0, 20.0, 20.0, 20.0, 20.0]  # [N,N,N,Nm,Nm.Nm]
      upper_force_thresholds_nominal: [40.0, 40.0, 40.0, 40.0, 40.0, 40.0]  # [N,N,N,Nm,Nm.Nm]
